wpilibj/src/main/java/org/wpilib/driverstation/GenericHID.java

```diff
@@ -129,6 +129,16 @@ public boolean getRawButton(int button) {
     return DriverStation.getStickButton(m_port, (byte) button);
   }
 
+  public boolean getRawButton(GamepadIndexPair button) {
+    int index;
+    if (DriverStation.isNiDs()) {
+      index = button.getNiIndex();
+    } else {
+      index = button.getSdlIndex();
+    }
```

Comment:
1. What should happen if the DS didn't connect yet? I have a feeling this is going to cause a lot of issues, similar to the alliance color issues in the past.
2. Is defaulting to SDL the desired behavior?
3. This evaluation should be pulled up to somewhere, rather than littering it all over the code, for consistency (in defaults etc) and code dedup.